### PR TITLE
Make app run at startup on mac

### PR DESCRIPTION
**Pull request by mbed67**
This PR is for the functionality on MacOS.

Todo:
- write logic for linux and windows

### CODE REVIEW

**Comment by SunyataZero** [comment]
@mbed67 I won't say i'm sorry, but i'll say that i wish i was able to both meet my need for connecting with you all, and my need for efficiency as well as my need for self-care. Now that i feel a bit better i looked at this and i think it looks good. I like the placement of the checkbox, it seems "logical" for me. Another option i've been thinking about is to move things into the main menu of the settings window, but maybe this can be done later (or not at all - if the settings window is maximized even on smaller computer screens we still have space left. Perhaps it's more of an issue of hiding things which are less important but so far everything we have in the main settings window is probably pretty important)

File: mc/gui/general_settings_wt.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import mc.mc_global
+import mc.model
+import os
+from PyQt5 import QtWidgets
+from shutil import copyfile
+from sys import platform
```
Comment (SunyataZero) [inline mc/gui/general_settings_wt.py]: We can also get the platform from Qt, right now we have "kernel type: linux". I'm not sure though which of these options (getting the platform from Qt or from Python) is safest

File: mc/gui/general_settings_wt.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class RunOnStartupWt(QtWidgets.QWidget):
+    def __init__(self):
+        super().__init__()
+        self.show()
+        self.run_on_startup = QtWidgets.QCheckBox(self.tr("Run on startup"))
```
Comment (SunyataZero) [inline mc/gui/general_settings_wt.py]: Minor thing: I like to add `_qcb` or similar to the end of widgets, where q is for Qt and cb (or something else) is an abbreviation of the widget type

**Comment by mbed67** [comment]
@SunyataZero I saw that you introduced the new options menu in another PR. Perhaps, when that is merged, it can be used to store this option. For now I made the checkbox visible only on mac.

**Comment by SunyataZero** [comment]
Great!

**Pull request merged**